lib/adapters/parsers/csv.js

```diff
@@ -22,10 +28,19 @@ module.exports = base.extend({
             });
 
             csvStream.on('data', function(pt){
-                buffer.push(pt);
-                if (buffer.length >= self.limit) {
-                    emit(buffer);
-                    buffer = [];
+                self.totalParsed++;
+
+                if (self.stopAt !== -1 && self.totalParsed > self.stopAt) {
+                    if (!self.doneParsing) { 
+                        self.doneParsing = true;
+                        this.emit('end');
```

Comment:
`this.emit` is on the csvStream not the parser class which I'll investigate why exactly this doesn't stop everything instead of continuing to parse the inbound stream.